arb_os/gasAccounting.mini

```diff
@@ -798,7 +808,12 @@ impure func payForTxAndL1Gas(
         fullAmount
     };
 
-    let payableToFee = fixedPoint_mulByUintTrunc(feeConfig.shareForNetFee, amount);
+    let payableToFee = xif(preferredAggregator == recipient) {
+        fixedPoint_mulByUintTrunc(feeConfig.shareForNetFee, amount)
+    } else {
+        amount
+    };
```

Comment:
I'm not sure I understand why we're only incorporating the network fee here if using the preferred aggregator. I assume is has to do with the adjustment we made if you're not using the preferred aggregator above, but this is confusing and needs some refactoring or documentation

Reply:
Added comments to explain this.